### PR TITLE
Feature/encapsulating

### PR DESCRIPTION
**Pull request by luqp**

### CODE REVIEW

File: src/Bank/BankLogic/AccountSettings/Owner.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        private readonly int fee;
+
+        private Owner(AccountOrigin origin, int fee)
+        {
+            Origin = origin;
+            this.fee = fee;
+        }
+
+        // Is it good in this way?
+        public static IOwnerStrategy GetOriginOwner(AccountOrigin origin) =>
```
Comment (uabarahona) [inline src/Bank/BankLogic/AccountSettings/Owner.cs]: Make an extension methods for this

File: src/Bank/BankLogic/AccountSettings/Owner.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace BankingSystem.BankLogic
+{
+    public class Owner : IOwnerStrategy
```
Comment (uabarahona) [inline src/Bank/BankLogic/AccountSettings/Owner.cs]: Rename the interface to get rid of "Strategy" part

File: src/Bank/BankLogic/BankAccount.cs [hunk 3]
```diff
@@ -4,16 +4,16 @@ namespace BankingSystem.BankLogic
 {
     public abstract class BankAccount : IBankAccount
     {
-        protected BankAccount(int accountNumber, AccountOrigin accountOrigin, double interestRate)
+        protected BankAccount(int accountNumber, AccountOrigin origin, double interestRate)
         {
             Balance = 0;
             AccountNumber = accountNumber;
-            AccountOrigin = accountOrigin;
+            OriginOwner = Owner.GetOriginOwner(origin);
```
Comment (uabarahona) [inline src/Bank/BankLogic/BankAccount.cs]: this should be at least `origin.GetOwner() `

Comment (uabarahona) [reply src/Bank/BankLogic/BankAccount.cs]: also rename the variable to be owner

File: src/Bank/BankLogic/Factories/IAccountsFactory.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace BankingSystem.BankLogic
+{
+    interface IAccountsFactory
+    {
+        IBankAccount Create(int accountNumber, AccountOrigin origin);
+
+        static IBankAccount CreateAccount(AccountType type, AccountOrigin origin, int accountNumber)
```
Comment (uabarahona) [inline src/Bank/BankLogic/Factories/IAccountsFactory.cs]: Please simplify this factory pattern

File: src/Bank/BankLogic/AccountSettings/Owner.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            Origin = origin;
+            this.fee = fee;
+        }
+
+        // Is it good in this way?
+        public static IOwnerStrategy GetOriginOwner(AccountOrigin origin) =>
+            origin switch
+            {
+                AccountOrigin.FOREIGN => new Owner(origin, 5),
+                AccountOrigin.LOCAL => new Owner(origin, 0),
```
Comment (uabarahona) [inline src/Bank/BankLogic/AccountSettings/Owner.cs]: _Please cache this default objects

File: src/BankConsole/Commands/CommandFactory.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            new ShowCmd(),
+            new AddInterest()
+        };
+
+        public ICommand GetCommand(int commandNumber)
+        {
+            return commands[commandNumber];
+        }
+
+        public string[] cmdNames()
```
Comment (uabarahona) [inline src/BankConsole/Commands/CommandFactory.cs]: please do not simplify the names

File: src/BankConsole/Commands/CommandFactory.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            List<string> names = new List<string>();
+            foreach( var cmd in commands)
+            {
+                names.Add(cmd.ToString());
+            }
+            return names.ToArray();
```
Comment (uabarahona) [inline src/BankConsole/Commands/CommandFactory.cs]: use the LINQ h ere to convert command to names

**Review by uabarahona — CHANGES_REQUESTED** [review]

File: src/Bank/BankLogic/AccountSettings/Owner.cs (hunk 1 above)
Comment (luqp) [reply src/Bank/BankLogic/AccountSettings/Owner.cs]: Ok, this is already

File: src/Bank/BankLogic/AccountSettings/Owner.cs (hunk 2 above)
Comment (luqp) [reply src/Bank/BankLogic/AccountSettings/Owner.cs]: I hope what "IAccountOrigin" as name will be fine

File: src/Bank/BankLogic/BankAccount.cs (hunk 3 above)
Comment (luqp) [reply src/Bank/BankLogic/BankAccount.cs]: ;)

File: src/Bank/BankLogic/Factories/IAccountsFactory.cs (hunk 4 above)
Comment (luqp) [reply src/Bank/BankLogic/Factories/IAccountsFactory.cs]: ok, I deleted the noise

File: src/Bank/BankLogic/AccountSettings/Owner.cs (hunk 5 above)
Comment (luqp) [reply src/Bank/BankLogic/AccountSettings/Owner.cs]: ok, I used a Dictionary to do

File: src/BankConsole/Commands/CommandFactory.cs (hunk 6 above)
Comment (luqp) [reply src/BankConsole/Commands/CommandFactory.cs]: well, this is my weakest skill

File: src/BankConsole/Commands/CommandFactory.cs (hunk 7 above)
Comment (luqp) [reply src/BankConsole/Commands/CommandFactory.cs]: Ok, thanks for reviews

**Review by uabarahona — APPROVED** [review]

**Pull request merged**